### PR TITLE
Peek and receive on queues for AzureServiceBus

### PR DESCRIPTION
**Pull request by Krusty93**

### CODE REVIEW

File: src/Server/EventBusExplorer.Server.API/Controllers/QueuesController.cs [hunk 1]
```diff
@@ -73,7 +80,34 @@ public async Task<IActionResult> GetAsync([FromRoute] string name)
...
+    /// </summary>
+    /// <param name="queueName">Queue name</param>
+    /// <param name="receiveMode">Receive mode</param>
+    /// <param name="subQueue">Sub queue to query</param>
+    /// <param name="fromSequenceNumber">(Optional) Fetch messages from this one</param>
+    /// <param name="cancellationToken">(Optional) Cancellation token to cancel the operation</param>
+    /// <response code="200">List of peeked messages</response>
+    [ProducesResponseType(typeof(GetMessagesResponse), StatusCodes.Status200OK)]
+    [HttpGet("{queueName}/messages")]
+    public async Task<IActionResult> PeekMessagesAsync(
```
Comment (matteopessina) [inline src/Server/EventBusExplorer.Server.API/Controllers/QueuesController.cs:98]:
```suggestion
    public async Task<IActionResult> GetMessagesAsync(
```
Per essere coerenti con quanto sotto e agnostici rispetto alla receiveMode specificata.

File: src/Server/EventBusExplorer.Server.API/Controllers/TopicsController.cs [hunk 2]
```diff
@@ -13,11 +13,18 @@ namespace EventBusExplorer.Server.API.Controllers;
 [Produces(MediaTypeNames.Application.Json)]
 public class TopicsController : ControllerBase
 {
-    private readonly IServiceBrokerTopicsService _topicService;
+    private readonly IEventBusManagementService _eventBusManagementService;
```
Comment (matteopessina) [inline src/Server/EventBusExplorer.Server.API/Controllers/TopicsController.cs:16]: I think we need to talk about the names with use for the concepts involved: are we talking about events or messages?
Probably a call would better fit this need.

File: src/Server/EventBusExplorer.Server.Application/Models/QuerySettings.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Diagnostics;
```
[the remark this answers is not in the document]
Comment (Krusty93) [reply src/Server/EventBusExplorer.Server.Application/Models/QuerySettings.cs:1]: Used by `[DebuggerDisplay]` attribute

File: src/Server/EventBusExplorer.Server.Application.ServiceBroker.Abstractions/Models/MessageListModel.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    long SequenceNumber,
+    string Subject,
+    string Body);
+
+public record MessageListModel
+{
+    public MessageListModel(IEnumerable<MessageModel> messages)
+    {
+        if (messages is null)
+            throw new ArgumentNullException(nameof(messages));
```
[the remark this answers is not in the document]
Comment (Krusty93) [reply src/Server/EventBusExplorer.Server.Application.ServiceBroker.Abstractions/Models/MessageListModel.cs:13]: `null`, not empty

File: src/Server/EventBusExplorer.Server.API/Controllers/QueuesController.cs [hunk 5]
```diff
@@ -13,11 +13,18 @@ namespace EventBusExplorer.Server.API.Controllers;
 [Produces(MediaTypeNames.Application.Json)]
 public class QueuesController : ControllerBase
 {
-    private readonly IServiceBrokerQueuesService _queueService;
+    private readonly IEventBusManagementService _eventBusManagementService;
+    private readonly IMessagesService _messagesService;
```
[the remark this answers is not in the document]
Comment (matteopessina) [reply src/Server/EventBusExplorer.Server.API/Controllers/QueuesController.cs:17]: The suggested change has been moved to a separated issue #34

File: src/Server/EventBusExplorer.Server.API/Controllers/TopicsController.cs (hunk 2 above)
Comment (matteopessina) [reply src/Server/EventBusExplorer.Server.API/Controllers/TopicsController.cs:16]: The suggested change has been moved to a separated issue #34

File: src/Server/EventBusExplorer.Server.Application/Models/QuerySettings.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (matteopessina) [reply src/Server/EventBusExplorer.Server.Application/Models/QuerySettings.cs:1]: I meant the file location.

[the remark this answers is not in the document]
Comment (matteopessina) [reply src/Server/EventBusExplorer.Server.Application/Models/QuerySettings.cs:1]: We agreed on moving the file out of Models folder and to remove the Models folder for now.

File: src/Server/EventBusExplorer.Server.Application/Responses.cs [hunk 6]
```diff
@@ -0,0 +1,5 @@
+﻿namespace EventBusExplorer.Server.Application;
```
[the remark this answers is not in the document]
Comment (matteopessina) [reply src/Server/EventBusExplorer.Server.Application/Responses.cs:1]: Agreed on leaving Requests.cs and Responses.cs DTO pattern.

**Review by matteopessina — APPROVED** [review]

**Pull request merged**